data/src/main/scala/cats/data/IList.scala

```diff
@@ -0,0 +1,276 @@
+package cats.data
+
+import algebra.Monoid
+import cats.Fold.{Continue, Return}
+import cats._
+
+import scala.annotation.tailrec
+
+/**
+ * Purely functional single linked list
+ * IList stands for Invariant List as opposed to [[scala.List]] which is covariant
+ */
+sealed abstract class IList[A] extends Product with Serializable {
+  import cats.data.IList._
+
+  /** add an [[IList]] to the front */
```

Comment:
Most readers expect to read `as ::: bs` as "put the *as* first, then the *bs*". I think we should have `as concat bs` be read similarly (prepend the `as` on the `bs).

Right now `as concat bs` reads as "put the *bs* first then the *as*".